src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -0,0 +1,794 @@
+      // No existing application was found in the database,
+      // or an error occurred while attempting to load the application
+      setApplicationID(null);
+      setEditorUsername((prevEditorUsername) => prevEditorUsername ?? userProfile.AD_Username);
+      setApplicants((prevApplicants) =>
+        prevApplicants.length ? prevApplicants : [{ Profile: userProfile, OffCampusProgram: '' }],
+      );
+      setUnsavedChanges(true);
+    };
+
+    if (!applicationID || !applicationDetails) {
+      setLoading(true);
+      // Check if the current user is on an application. Returns the application ID number if found
+      housing
+        .getCurrentApplicationID()
+        .then((newApplicationID) => {
+          console.log('Retrieved Application ID: ' + newApplicationID); //! DEBUG
+          if (newApplicationID === null || newApplicationID === -1) {
+            // Intentionally trigger the 'catch'
+            throw new Error("Invalid value of 'newApplicationID' = " + newApplicationID);
+          } else if (isSubscribed) {
+            setApplicationID(newApplicationID);
+            setUnsavedChanges(false);
+            housing
+              .getApartmentApplication(newApplicationID)
+              .then((newApplicationDetails) => {
+                if (isSubscribed && newApplicationDetails) {
+                  setApplicationDetails(newApplicationDetails);
+                  setDateSubmitted(newApplicationDetails.DateSubmitted ?? null);
+                  setDateModified(newApplicationDetails.DateModified ?? null);
+                  setEditorUsername(newApplicationDetails.EditorUsername ?? null);
+                  setApplicants((prevApplicants) => {
+                    if (newApplicationDetails.Applicants?.length > 0) {
+                      return newApplicationDetails.Applicants.map(async (newApplicantInfo) => {
+                        if (newApplicantInfo.Profile !== null) {
+                          return newApplicantInfo;
+                        } else {
+                          const newApplicantProfile = await user.getProfileInfo(
+                            newApplicantInfo.Username,
+                          );
+                          return { Profile: newApplicantProfile, ...newApplicantInfo };
+                        }
+                      });
+                    } else {
+                      return prevApplicants;
+                    }
+                  });
+                  setPreferredHalls(newApplicationDetails?.ApartmentChoices ?? []);
+                  setUnsavedChanges(false);
+                }
+              })
+              .catch((error) => {
+                handleError(error);
+              });
+          }
+        })
+        .catch((error) => {
+          handleError(error);
+        });
+      setLoading(false);
+    }
+
+    return () => (isSubscribed = false);
+  }, [userProfile, applicationDetails, applicationID, editorUsername, applicants]);
```

Comment:
You only want to re-run this useEffect when userProfile changes. The rest of these should not be dependencies of this effect.